packages/story-editor/src/utils/isTargetCoveringContainer.js

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+function isTargetCoveringContainer(target, container) {
+  const targetBox = target.getBoundingClientRect();
+  const containerBox = container.getBoundingClientRect();
+
+  const intersectionArea =
```

Comment:
Hm, wondering about the rotated media, does this consider rotation as well?

We're also using [`sat`](https://www.npmjs.com/package/sat) library in the editor (search `import SAT from 'sat';` for use cases) which might have something that could calculate this for us, e.g. there seems to be `aInB` which says if polygon A is fully inside polygon B -- might be worthwhile checking.

Reply:
The question is, how should it work? because there is no rotation in the page background element, so maybe rotated images shouldn't be auto-set 🤔

Reply:
The more I'm thinking about it the more I'm convinced that it shouldn't autoset rotated images. Imagine someone carefully resizing the image to cover the whole canvas and then realizing that the image is straightened up.
@aaskedall @agingoldseco WDYT?

Reply:
My ux gut is that we probably should have a way for users to rotate a bg image once it's set as bg. This would then allow for a user to set a rotated image as bg. What do you think about adding this feature?

Reply:
Agree, we will probably also want to add BG color when media is set (I saw the discussion about it somewhere?). So it will take a while, until then, we can:
1) leave it as is 
2) prevent auto-set for rotated media

You can check how it works now on [QA env](https://stories-qa-wordpress-amp.pantheonsite.io/).

Reply:
> The more I'm thinking about it the more I'm convinced that it shouldn't autoset rotated images. Imagine someone carefully resizing the rotated image to cover the whole canvas and then realizing that the image is straightened up.

Makes sense! Especially if the plan is to support rotated background as well later.

Reply:
IIRC in the past we deliberately decided to not support rotated background elements due to potential issues that could arise from that. Just to keep that in mind when creating a ticket for adding support for this. Personally I've never seen any user requests for this, so perhaps we could even save us this hassle :-)

Reply:
To go back to @miina's original comment, I think using `sat` here makes sense just to avoid stupid math mistakes - we already have the lib loaded anyway.

Reply:
@barklund @miina 
I really wanted to use this `Separating Axis Theorem` 😎  
I did the implementation, but for me, it's too easy to create a state where it doesn't detect [`aInB`](https://github.com/jriecken/sat-js#satresponse), that's probably because our snapping is not pixel perfect (and that's why I have `coverRatio > 0.995`). `overlap...` props don't provide anything helpful in this case (it's just one [shortest colliding] axis info).

![image](https://user-images.githubusercontent.com/21036224/152997333-ee077f40-32a3-4435-a8a2-6260a2bdb2d8.png)

![image](https://user-images.githubusercontent.com/21036224/152994571-8a0ca68d-b67d-4caf-abaa-5c10e8c6300d.png)

Reply:
@merapi Could we solve this issue by leaving a buffer when creating a polygon to compare?
E.g. `const buffer = 2;` and then when creating the page area polygon, deduct/add `buffer` to the points? Or would rotation mess this up?

Reply:
Hm, yeah, I see the point. I'm also okay with keeping this as-is. It is not the hardest math ever. Probably won't win you a Fields Medal anytime soon.

Reply:
Hm, odd. We're also using it for the lasso selection, is that also not working correctly then? 🤔 
Probably need a bug issue for investigating. Perhaps `pointInPolygon` (used for layer selection) is not working as expected with rotation but `testPolygonPolygon` (used for lasso) is. Or maybe the library is somehow incorrectly used for layer selection.

Reply:
I've quickly checked the lasso - seems fine, but layer selection is indeed broken when dealing with rotated objects.
Small debug tool (transparent area - used in SAT calculations, dashed box - regular bb):

https://user-images.githubusercontent.com/21036224/153068858-0159d4a8-b1c2-40d1-94e1-3305ad501c5c.mov